go-controller/pkg/ovn/gress_policy.go

```diff
@@ -74,8 +75,12 @@ func newGressPolicy(policyType knet.PolicyType, idx int, namespace, name, portGr
 }
 
 func (gp *gressPolicy) addPortPolicy(portJSON *knet.NetworkPolicyPort) {
+	protocol := v1.ProtocolTCP
+	if portJSON.Protocol != nil {
+		protocol = *portJSON.Protocol
+	}
```

Comment:
Should not be necessary; apiserver defaulting should have filled that in.
(If it's only needed for the unit tests then fix it there.)